systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java

```diff
@@ -174,6 +183,179 @@ void testJSONFormatLogging() {
+                        "appender.console.name=STDOUT\n" +
+                        "appender.console.layout.type=PatternLayout\n" +
+                        "appender.console.layout.pattern=[%d] %-5p <%-12.12c{1}:%L> [%-12.12t] %m%n\n" +
+                        "rootLogger.level=OFF\n" +
+                        "rootLogger.appenderRefs=stdout\n" +
+                        "rootLogger.appenderRef.console.ref=STDOUT\n" +
+                        "rootLogger.additivity=false"))
+                .build();
+
+        kubeClient().getClient().configMaps().inNamespace(NAMESPACE).createOrReplace(configMapTo);
+        kubeClient().getClient().configMaps().inNamespace(NAMESPACE).createOrReplace(configMapUo);
+
+        ExternalLogging elTo = new ExternalLoggingBuilder()
+                .withName("external-configmap-to")
+                .build();
+
+        ExternalLogging elUo = new ExternalLoggingBuilder()
+                .withName("external-configmap-uo")
+                .build();
+
+        LOGGER.info("Setting log level of TO and UO to OFF - records should not appear in log");
+        // change to external logging
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            k.getSpec().getEntityOperator().getTopicOperator().setLogging(elTo);
+            k.getSpec().getEntityOperator().getUserOperator().setLogging(elUo);
+        });
+
+        LOGGER.info("The EO shouldn't roll - verifying pod stability");
+        DeploymentUtils.waitForNoRollingUpdate(KafkaResources.entityOperatorDeploymentName(CLUSTER_NAME), eoPods);
+
+        LOGGER.info("Waiting for log4j2.properties will contain desired settings");
+        TestUtils.waitFor("Logger change", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> cmdKubeClient().execInPodContainer(eoPodName, "topic-operator", "cat", "/opt/topic-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=OFF")
+                        && cmdKubeClient().execInPodContainer(eoPodName, "user-operator", "cat", "/opt/user-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=OFF")
+                        && cmdKubeClient().execInPodContainer(eoPodName, "topic-operator", "cat", "/opt/topic-operator/custom-config/log4j2.properties").out().contains("monitorInterval=30")
+                        && cmdKubeClient().execInPodContainer(eoPodName, "user-operator", "cat", "/opt/user-operator/custom-config/log4j2.properties").out().contains("monitorInterval=30")
+        );
+
+        LOGGER.info("Waiting {} ms for DEBUG log will disappear", LOGGING_RELOADING_INTERVAL);
+        Thread.sleep(LOGGING_RELOADING_INTERVAL * 2);
+
+        LOGGER.info("Asserting if log is without records");
+        assertThat(StUtils.getLogFromPodByTime(eoPodName, "topic-operator", "1m"), is(emptyString()));
+        assertThat(StUtils.getLogFromPodByTime(eoPodName, "user-operator", "1m"), is(emptyString()));
```

Comment:
I think I still don't get this. You update the configuration and wait 2*30 seconds for the log reload. In the worst case, the reload happens at the end of the first 30 seconds and the new settings will be valid for the second 30 seconds. So checking the logs for last minut still does not seem to work for me. The same on the line 232.